### PR TITLE
Move to zio framework, add startdate and duration params

### PR DESCRIPTION
**Pull request by saucam**
This PR continues to further consolidate the base framework for the project as part of #1 
Here we move to [Zio](https://github.com/zio/zio) to make the code purely functional.
Some immediate benefits include
- Ability to test every module
- Some essential constructs like ```ensuring``` or ```brackets``` that make it easier to further develop the spark job

Have also added 2 more command line params ```startdate``` and ```duration```
Moved the spark parameters to the bash script so we can trigger spark-submit based on target envrionment

### CODE REVIEW

File: src/main/scala/com/astrolabsoftware/grafink/CLParser.scala [hunk 1]
```diff
@@ -42,7 +49,33 @@ trait CLParser {
...
+          if (validateDate(x)) success
+          else failure("Option --startdate must be a valid date in format (yyyy-MM-dd)")
+        )
+        .text("startdate accepts start day's date for which the job needs to be run (yyyy-MM-dd)")
+
+      opt[Int]('d', "duration")
+        .optional()
+        .action((x, c) => c.copy(duration = x))
+        .validate(x =>
+          if (x > 0 && x <= 30) success
```
Comment (JulienPeloton) [inline src/main/scala/com/astrolabsoftware/grafink/CLParser.scala]: What is driving the max value for `duration`?

**Review by JulienPeloton — APPROVED** [review]
Thanks, it looks good to me! I did not know ZIO yet, so it took me some time to explore it - nice!

File: src/main/scala/com/astrolabsoftware/grafink/CLParser.scala (hunk 1 above)
Comment (saucam) [reply src/main/scala/com/astrolabsoftware/grafink/CLParser.scala]: Just the maximum amount of data that we want to process at a given time ?

Comment (JulienPeloton) [reply src/main/scala/com/astrolabsoftware/grafink/CLParser.scala]: OK - so probably a week is more than enough (for LSST, we would have about a TB per night).

Comment (saucam) [reply src/main/scala/com/astrolabsoftware/grafink/CLParser.scala]: ok , let me change this

Comment (saucam) [reply src/main/scala/com/astrolabsoftware/grafink/CLParser.scala]: > Thanks, it looks good to me! I did not know ZIO yet, so it took me some time to explore it - nice!

I am exploring it properly myself :)

**Comment by JulienPeloton** [comment]
All good - thanks for the new code. Merging!

**Pull request merged**